### PR TITLE
8273616: Fix trivial doc typos in the java.base module

### PR DESCRIPTION
**Pull request by pavelrappo**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8273616](https://bugs.openjdk.java.net/browse/JDK-8273616): Fix trivial doc typos in the java.base module

### Reviewers
 * [John R Rose](https://openjdk.java.net/census#jrose) (@rose00 - **Reviewer**)
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - **Reviewer**)
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**)
 * [Roger Riggs](https://openjdk.java.net/census#rriggs) (@RogerRiggs - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5475/head:pull/5475` \
`$ git checkout pull/5475`

Update a local copy of the PR: \
`$ git checkout pull/5475` \
`$ git pull https://git.openjdk.java.net/jdk pull/5475/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5475`

View PR using the GUI difftool: \
`$ git pr show -t 5475`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5475.diff">https://git.openjdk.java.net/jdk/pull/5475.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/nio/channels/AsynchronousFileChannel.java [hunk 1]
```diff
@@ -396,7 +396,7 @@ public static AsynchronousFileChannel open(Path file, OpenOption... options)
      * @param   metaData
      *          If {@code true} then this method is required to force changes
      *          to both the file's content and metadata to be written to
-     *          storage; otherwise, it need only force content changes to be
+     *          storage; otherwise, it needs only force content changes to be
```
Comment (rose00) [inline src/java.base/share/classes/java/nio/channels/AsynchronousFileChannel.java:399]: This is a correct though rare use of subjunctive mood:  One need not change it.
However, if be desired that we change the mood to indicative, I suggest s/it need only force/it only needs to force/.

File: src/java.base/share/classes/java/nio/channels/FileChannel.java [hunk 2]
```diff
@@ -564,7 +564,7 @@ public final long write(ByteBuffer[] srcs) throws IOException {
      * @param   metaData
      *          If {@code true} then this method is required to force changes
      *          to both the file's content and metadata to be written to
-     *          storage; otherwise, it need only force content changes to be
+     *          storage; otherwise, it needs only force content changes to be
```
Comment (rose00) [inline src/java.base/share/classes/java/nio/channels/FileChannel.java:567]: (same as previous comment:  the suggested fix makes the English *less* correct)

**Review by rose00 — APPROVED** [review]
Approved, except for two changes commented above: the original "it need only force" is correct usage, and "it needs only force" is not good usage, but "it only needs to force" would be clearer and also correct.

File: src/java.base/share/classes/java/nio/channels/AsynchronousFileChannel.java (hunk 1 above)
Comment (efge) [reply src/java.base/share/classes/java/nio/channels/AsynchronousFileChannel.java:399]: (`it need only force` is not really subjunctive mood, but rather usage as a modal auxiliary, see https://english.stackexchange.com/a/297235)

File: src/java.base/share/classes/java/nio/channels/FileChannel.java (hunk 2 above)
Comment (pavelrappo) [reply src/java.base/share/classes/java/nio/channels/FileChannel.java]: Reverted both in 9a9deee; thanks.

**Review by irisclark — APPROVED** [review]

**Review by LanceAndersen — APPROVED** [review]

**Comment by pavelrappo** [comment]
Thanks to those who reviewed this PR. Since I posted it, I've found three more occurrences of _insure_ used in the sense of _ensure_: two in the `java.io.Object*Stream` area and one in the `java.util.Currency` class. I decided to fix these typos in this PR, which now needs to be (re)reviewed. Thanks!

There are more occurrences of _insure_, which I didn't touch. Some of them are in java.sql, java.sql.rowset and java.desktop. In the latter, _insure_ even crept into method names.

**Review by dfuch — APPROVED** [review]
LGTM

**Review by RogerRiggs — APPROVED** [review]

**Review by irisclark — APPROVED** [review]

**Review by LanceAndersen — APPROVED** [review]

**Review by rose00 — APPROVED** [review]

**Pull request closed without merging**